src/routers/handlers/enter_your_details/index.ts

```diff
@@ -0,0 +1,100 @@
+import { setPresenterAccountDetails, getPresenterAccountDetails, fetchUserDetails } from "./../../../utils/session";
+import { ValidationError, validationResult } from "express-validator";
+import { ErrorManifestValidationType } from "utils/error_manifests/default";
+
+interface CountryOptions {
+    value: string,
+    text: string,
+    selected?: boolean
+}
+
+interface EnterYourDetailsViewData extends BaseViewData{
+    address: Address ;
+    countries: CountryOptions[];
+}
+
+export class EnterYourDetailsHandler extends GenericHandler<EnterYourDetailsViewData>{
+    private static readonly templatePath = "router_views/enter-your-details/enter-your-details";
+    readonly title = "What is your correspondence address?";
+
+    /**
+     * Gets the viewdata from the request
+     * @param req : Request
+     * @returns: EnterYourDetailsViewData
+     */
+    public getViewData(req: Request): EnterYourDetailsViewData {
+        const baseViewData = super.getViewData(req);
+        // retrieve details using session key
+        const details = getPresenterAccountDetails(req);
+
+        if (details.userId === undefined) {
+            throw new Error("Presenter account details not found in session.");
+        }
+
+        return {
+            ...baseViewData,
+            title: this.title,
+            backURL: PrefixedUrls.APPLY_TO_FILE_OPTIONS,
+            address: details?.address || req.body?.address,
+            countries: [{ value: 'Select a country', text: 'Select a country', selected: true }, ...countries]
+        };
+    }
+
+    public executeGet(req: Request, _response: Response): ViewModel<EnterYourDetailsViewData>{
+        logger.info(`${this.constructor.name} get execute called`);
+
+        const viewData = this.getViewData(req);
+
+        return {
+            templatePath: EnterYourDetailsHandler.templatePath,
+            viewData
+        };
+    }
+
+    public executePost(req: Request, _response: Response): ViewModel<EnterYourDetailsViewData> | Redirect {
+        logger.info(`${this.constructor.name} post execute called`);
+        // validating the form using the req object
+        const errors = validationResult(req);
+
+        if (!errors.isEmpty()){
+            const viewData = this.getViewData(req);
+            // if validation errors exists, get them as an array
+            viewData.errors = this.convertValidationErrorsToErrorManifestType(errors.array());
+            return {
+                templatePath: EnterYourDetailsHandler.templatePath,
+                viewData
+            };
+        }
+        let details =  getPresenterAccountDetails(req);
+        details.address = { ...req.body };
+
+        if (details.userId === undefined) {
+            details = fetchUserDetails(req, details);
+        }
+
+        setPresenterAccountDetails(req, details);
+        // generate the redirect query string
+        const redirect = PrefixedUrls.CHECK_DETAILS;
+        return { redirect };
```

Comment:
There is no need tto store this in a temporary varaible just to immediately put it in an object.
Just return an object literal.
```suggestion
        return { 
            redirect: PrefixedUrls.CHECK_DETAILS
        };
```